### PR TITLE
style: Unify needs_traversal logic.

### PR DESCRIPTION
**Pull request by emilio**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17782)

### CODE REVIEW

File: components/style/traversal.rs [hunk 1]
```diff
@@ -267,10 +266,15 @@ pub trait DomTraversal<E: TElement> : Sync {
     fn element_needs_traversal(
         el: E,
         traversal_flags: TraversalFlags,
-        data: &ElementData,
+        data: Option<&ElementData>,
         parent_data: Option<&ElementData>,
     ) -> bool {
-        debug_assert!(data.has_styles(), "Caller should check this");
+        debug!("element_needs_traversal({:?}, {:?}, {:?}, {:?})",
+               el, traversal_flags, data, parent_data);
```
Comment (heycam) [inline components/style/traversal.rs]: Nit: maybe blank line after this?

**Comment by emilio** [comment]
@bors-servo r-

Need to be away for a bit and forgot to push to geckotry without the other patches, so will do that.

**Pull request merged**